core/coreapi/interface/options/unixfs.go

```diff
@@ -42,7 +42,13 @@ type UnixfsAddSettings struct {
 	Progress bool
 }
 
+type UnixfsLsSettings struct {
+	ResolveType bool
+	ResolveSize bool
```

Comment:
As a side effect of both those operations (resolution of child nodes), they don't need to be separate options. If any of them is true the work is done, so there is no reason to have them both.

IMO have one `ResolveChildren` option.